sale_order_revision/models/sale_order.py

```diff
@@ -34,11 +34,37 @@ def _prepare_revision_data(self, new_revision):
 
     def action_view_revisions(self):
         self.ensure_one()
-        result = self.env["ir.actions.actions"]._for_xml_id("sale.action_orders")
-        result["domain"] = ["|", ("active", "=", False), ("active", "=", True)]
-        result["context"] = {
+        domain = ["|", ("active", "=", False), ("active", "=", True)]
+        context = {
             "active_test": 0,
             "search_default_current_revision_id": self.id,
             "default_current_revision_id": self.id,
         }
-        return result
+        search_view = self.env.ref("sale.sale_order_view_search_inherit_sale")
+        return {
+            "name": _("Previous Revisions"),
+            "view_mode": "tree,form",
+            "res_model": "sale.order",
+            "type": "ir.actions.act_window",
+            "search_view_id": search_view.id,
+            "domain": domain,
+            "context": context,
+        }
+
+    def create_revision(self):
+        # Extends base_revision module
+        action = super().create_revision()
+        # Keep links to Invoices on the new Sale Order
+        old_lines = self.order_line
+        new_lines = self.current_revision_id.order_line
+        for old_line, new_line in zip(old_lines, new_lines):
+            new_line.invoice_lines = old_line.invoice_lines
+        return action
+
+    def action_cancel_create_revision(self):
+        for sale in self:
+            sale.action_cancel()
+            action = sale.create_revision()
+        if len(self) == 1:
+            return action
+        return {}
```

Comment:
```suggestion
    def action_cancel_create_revision(self):
        self.ensure_one()
        sale.action_cancel()
        return sale.create_revision()
```

If you can only call from the form view for a record, I think it could be simplified like this.